app/src/main/java/com/imashnake/animite/features/media/MediaPageViewModel.kt

```diff
@@ -1,95 +1,93 @@
 
                 uiState = with(uiState) {
                     copy(
                         bannerImage = media?.bannerImage,
                         coverImage = media?.coverImage?.extraLarge,
-                        color = media?.coverImage?.color,
+                        color = media?.coverImage?.color?.let { Color.parseColor(it) },
                         title = media?.title?.romaji ?:
-                                media?.title?.english ?:
-                                media?.title?.native,
+                        media?.title?.english ?:
+                        media?.title?.native,
                         description = media?.description,
                         stats = score + ranks,
-                        genres = media?.genres,
+                        genres = media?.genres?.filterNotNull(),
                         characters = media?.characters?.nodes?.map {
                             Character(
                                 id = it?.id,
                                 image = it?.image?.large,
                                 name = it?.name?.full
                             )
                         },
-                        trailer = Trailer(
-                            link = when (media?.trailer?.site) {
-                                "youtube" -> {
-                                    "https://www.youtube.com/watch?v=${media.trailer.id}"
-                                }
-                                "dailymotion" -> {
-                                    "https://www.dailymotion.com/video/${media.trailer.id}"
-                                }
-                                else -> {
-                                    null
-                                }
-                            },
-                            thumbnail = when (media?.trailer?.site) {
-                                // TODO: Does a high resolution image always exist?
-                                "youtube" -> {
-                                    "https://img.youtube.com/vi/${media.trailer.id}/maxresdefault.jpg"
-                                }
-                                // TODO: Change the icon and handle this properly.
-                                "dailymotion" -> {
-                                    media.trailer.thumbnail
-                                }
-                                else -> {
-                                    null
-                                }
-                            }
-                        )
+                        trailer = media?.trailer?.toUiModel()
                     )
                 }
             } catch(ioe: IOException) {
                 TODO()
             }
         }
     }
+
+    private fun MediaQuery.Trailer.toUiModel(): Trailer? {
+        // Give up if we don't have the data we want
+        if (site == null || thumbnail == null || id == null) return null
+        return Trailer(
+            link = when (site) {
```

Comment:
TODO - probably worth making this an enum

Reply:
Added TODO to code